### PR TITLE
docs: mock guide

### PR DESCRIPTION
**Pull request by yury-s**
#11670

### CODE REVIEW

File: docs/src/mock-js.md [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+
+  await page.goto('/');
+  await expect(page.locator('.battery-percentage')).toHaveText('90%');
+  await expect(page.locator('.battery-status')).toHaveText('Adapter');
+  await expect(page.locator('.battery-fully')).toHaveText('00:30');
+})
+
+```
+
+## Golden tests
```
Comment (dgozman) [inline docs/src/mock-js.md:65]: `## Verifying API calls`?

File: docs/src/mock-js.md [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+Sometimes it is useful to check if the page made all expected APIs calls. You can
+record all API method invocations and then compare them with golden result.
+[`method: Page.exposeFunction`] may come in handy for passing message from
+the page back to the test code:
+
+```js
+test('log battery calls', async ({ page }) => {
+  const log = [];
+  // Expose function for pushing messages to the Node.js script.
+  await page.exposeFunction('pwlog', msg => log.push(msg));
```
Comment (dgozman) [inline docs/src/mock-js.md:76]: I'd call it `logCall`

File: docs/src/mock-js.md [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+      level: 0.75,
+      charging: true,
+      chargingTime: 1800,
+      dischargingTime: Infinity,
+      // Log addEventListener calls.
+      addEventListener: (name, cb) => pwlog(`addEventListener:${name}`)
+    };
+    // Override the method to always return mock battery info.
+    window.navigator.getBattery = async () => {
+      pwlog('getBattery')
```
Comment (dgozman) [inline docs/src/mock-js.md:88]: missing semicolon

File: docs/src/mock-js.md [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+      return mockBattery;
+    };
+    delete window.navigator.battery;
```
Comment (dgozman) [inline docs/src/mock-js.md:91]: Drop this one?

File: docs/src/mock-js.md [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+  });
+
+  await page.goto('/');
+  await expect(page.locator('.battery-percentage')).toHaveText('75%');
+
+  // Compare actual calls with golden.
```
Comment (dgozman) [inline docs/src/mock-js.md:97]: golden -> expected

File: docs/src/mock-js.md [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+        this._chargingListeners.forEach(cb => cb());
+      }
+    };
+    const mockBattery = new BatteryMock();
+    // Override the method to always return mock battery info.
+    window.navigator.getBattery = async () => mockBattery;
+    // Save the mock object on window for easier access.
+    window.mockBattery = mockBattery;
+
+    // application tries navigator.battery first
```
Comment (dgozman) [inline docs/src/mock-js.md:145]: Same here.

**Review by dgozman — APPROVED** [review]

File: docs/src/mock-js.md [hunk 7]
```diff
@@ -0,0 +1,163 @@
...
+    window.navigator.getBattery = async () => mockBattery;
+    // application tries navigator.battery first
+    // so we delete this method
+    delete window.navigator.battery
+  });
+
+  await page.goto('/');
+  await expect(page.locator('.battery-percentage')).toHaveText('90%');
+  await expect(page.locator('.battery-status')).toHaveText('Adapter');
+  await expect(page.locator('.battery-fully')).toHaveText('00:30');
```
[the remark this answers is not in the document]
Comment (yury-s) [reply docs/src/mock-js.md:60]: This is how it's called in the [actual example](https://github.com/pazguille/demo-battery-api), I'll add an example project in a follow-up.

**Pull request merged**